### DISCUSSION TITLE
Change Invalid character in default hostname

### DISCUSSION THREAD

**Discussion by thebhd** [arendst/Tasmota#11196] [Ideas]
### PROBLEM DESCRIPTION

On a newly flashed tasmota device the default hostname "%s-%04d" which translates to tasmota_xxxxxx-yyyy 
 
This contains an underscore which turns out to not be valid character in a hostname. Resulting in being unable to do a dns lookup for the device in windows 10 and Debian.

Whilst this issue is trivial to fix by changing the configuration, it will cause frustration to novice users and seriously frustrates efforts to automate management of a large fleet of devices.   

#### Suggested Fix

Change the underscore to a hyphen in the default mqtt topic (from: tasmota_%06X to: tasmota-%06X) 

Also ensure that the default SSID when the device is in AP mode continues to match the hostname (current behaviour) as this makes automated configuration/management  of new devices really easy.  

Note to new users of tasmota:  
%s in the hostname is the mqtt topic 

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): _____ Any esp8266 device 
- [x] Tasmota binary firmware version number used: _____ 9.1.0, 9.2.0 & 9.3.1
  - [x] Pre-compiled 
  - [ ] Self-compiled
- [x] Flashing tools used: esptool.py, 

- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
 N/A Affects all default configurations using the precompiled tasmota.bin firmware
```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
Erase and re-flash the device with fresh image - behaviour is by design

### EXPECTED BEHAVIOUR

nslookup/hosts to return a valid ip address - underscore in host name causes unpredictable behaviour and failure with 
some dns servers 

### SCREENSHOTS
N/A

### ADDITIONAL CONTEXT

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by jaymemaurice** [comment]
Reluctant +1 to "let's fix this?" maybe
I don't like that it's wrong but all my devices are %s-%04d with default topic names and friendlynames matching hostname-1 -2 etc.

I do believe when these ESP break or are flashing, they are ESP_XXXXX
